src/fields/Fields.H

```diff
@@ -44,16 +44,42 @@ static std::array<std::map<std::string, int>, 5> Comps
             }}
     }};
 
-/** \brief Operation performed in the TransverseDerivative function:
- * either assign or add to the destination array
- */
-enum struct SliceOperatorType { Assign, Add };
-
 /** \brief Direction of each dimension. Can be used for clean handling 2D vs. 3D in the future */
 struct Direction{
     enum dir{x=0, y, z};
 };
 
+/** \brief Helper class to pass non-owning, lightweight views of specific Fields around */
+struct FieldView {
+    // use brace initialization or getField/getStagingArea as constructor
+    amrex::MultiFab& m_mfab;
+    int m_comp;
+
+    /** \brief get amrex::Array4<const? amrex::Real> of the specific Field component
+     * \param[in] mfi MFIter object of this or a related field
+     */
+    auto array (const amrex::MFIter& mfi) const {
+        return m_mfab.array(mfi, m_comp);
+    }
+};
+
+/** \brief Function to help converting field indexes to positions and backwards. Usage:
+ * x = i * dx + GetPosOffset(0, geom, box)
+ * i = round( (x - GetPosOffset(0, geom, box)) / dx )
+ *
+ * \param[in] direction 0, 1 or 2 for x, y or z respectively
+ * \param[in] geom geometry of the field
+ * \param[in] box box of the filed, can be with or without ghost cells
+ */
+inline amrex::Real
+GetPosOffset (const int direction, const amrex::Geometry& geom, const amrex::Box& box) {
+    using namespace amrex::literals;
+    // match boxes at center point
+    return 0.5_rt * (geom.ProbLo(direction) + geom.ProbHi(direction)
+           - geom.CellSize(direction) * (box.smallEnd(direction) + box.bigEnd(direction)));
+}
```

Comment:
Just a note (no change required), `dir` is a well-established name, if you want to shorten `direction`.